Automattic/wp-calypso
# Userless checkout

#### Changes proposed in this Pull Request

* Implement "userless checkout" flow, as described in pau2Xa-1hW-p2. Here, we remove the user registration step in signup flow, and directly take users who select a paid plan to checkout. So, the checkout page loads "userless" - no user account is created yet at this point. The account and site gets created when the user clicks the "Pay" button. 

#### Testing instructions



* Apply D45659-code
* This works only on the new checkout, so ensure that you're on a US IP address in EN locale.
* While logged out, go to /start/onboarding-new to begin the userless checkout flow.
* Selecting a free plan will show the user registration step(since there won't be any checkout page). Selecting a paid plan will show the checkout page with an email field. 
* This works only for credit card payment for now.
